Tests/BuildPowerPlatform.Action.Test.ps1

```diff
@@ -0,0 +1,118 @@
+        $versionNodeText | Should -Not -Contain $oldVersionString
+        $versionNodeText | Should -Contain $newVersionString
+    }
+
+    It 'Updates the Power App connections' {
+        write-host "Run update BC connections test"
+        # note: The old company name and environment name are hardcoded in the test data
+        $oldCompanyName = "TestCompanyId"
+        $oldEnvironmentName = "TestEnvironmentName"
+
+        $newCompanyName = "NewCompanyName"
+        $newEnvironmentName = "NewEnvironmentName"
+
+        $solutionPath = "$testDataTempPath/StandardSolution";
+
+        # Check file content before running the script
+        # NOTE: There are multiple connection files in the test data, but we only check one of them as a smoke test
+        $connectionFileContent = [string](Get-Content -Encoding UTF8 -Path (Join-Path $solutionPath 'CanvasApps/src/TestApp/Connections/Connections.json'))
+        $connectionFileContent | Should -Not -BeNullOrEmpty
+        $connectionFileContent | Should -Match $oldCompanyName
+        $connectionFileContent | Should -Match $oldEnvironmentName
+        $connectionFileContent | Should -Not -Match $newCompanyName
+        $connectionFileContent | Should -Not -Match $newEnvironmentName
+
+        $workflowFileContent = [string](Get-Content -Encoding UTF8 -Path (Join-Path $solutionPath 'Workflows/TestWorkflow-ABA81736-12D9-ED11-A7C7-000D3A991110.json'))
+        $workflowFileContent | Should -Not -BeNullOrEmpty
+        $workflowFileContent | Should -Match $oldCompanyName
+        $workflowFileContent | Should -Match $oldEnvironmentName
+        $workflowFileContent | Should -Not -Match $newCompanyName
+        $workflowFileContent | Should -Not -Match $newEnvironmentName
+
+        # Run the script
+        BuildPowerPlatform -solutionFolder $solutionPath -CompanyId $newCompanyName -EnvironmentName $newEnvironmentName
+
+        # Check file content after running the script
+        $connectionFileContent = [string](Get-Content -Encoding UTF8 -Path (Join-Path $solutionPath 'CanvasApps/src/TestApp/Connections/Connections.json'))
+        $connectionFileContent | Should -Not -BeNullOrEmpty
+        $connectionFileContent | Should -Not -Match $oldCompanyName
+        $connectionFileContent | Should -Not -Match $oldEnvironmentName
+        $connectionFileContent | Should -Match $newCompanyName
+        $connectionFileContent | Should -Match $newEnvironmentName
+
+        $workflowFileContent = [string](Get-Content -Encoding UTF8 -Path (Join-Path $solutionPath 'Workflows/TestWorkflow-ABA81736-12D9-ED11-A7C7-000D3A991110.json'))
+        $workflowFileContent | Should -Not -BeNullOrEmpty
+        $workflowFileContent | Should -Not -Match $oldCompanyName
+        $workflowFileContent | Should -Not -Match $oldEnvironmentName
+        $workflowFileContent | Should -Match $newCompanyName
+        $workflowFileContent | Should -Match $newEnvironmentName
+    }
+
+    It 'Works with PowerApp Only' {
```

Comment:
We had some issues in the past, where the scripts that perform the value update would break if the solutions only contained a Power App or only contained Power Automate Flows. 

We just added these tests to verify that both scenarios are supported.